src/main/java/io/github/eocqrs/xfake/FkSynchronizedStorage.java

```diff
@@ -0,0 +1,51 @@
+package io.github.eocqrs.xfake;
+
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import org.xembly.Directive;
+
+import com.jcabi.xml.XML;
+
+import lombok.RequiredArgsConstructor;
+
+/**
+ * This is synchronized decorator for {@link FkStorage}.
+ * <b>This class is thread-safe.</b>
+ *
+ * @since 0.0.2
+ * @author l3r8yJ
+ */
+@RequiredArgsConstructor
+public final class FkSynchronizedStorage implements FkStorage {
```

Comment:
@l3r8yJ maybe simplify to just `Synchronized`?

WDYT?

Reply:
@h1alexbel I'm not sure that Java would accept such a name